### PR TITLE
ART-3507 Refactor ocp4-scan

### PR DESCRIPTION
**Pull request by locriandev**

### CODE REVIEW

File: doozer/doozerlib/cli/scan_sources.py [hunk 1]
```diff
@@ -3,126 +3,205 @@
...
+                    self.add_image_meta_change(meta, rebuild_hint)
 
             else:
                 raise IOError(f'Unsupported meta type: {meta.meta_type}')
 
-        # To limit the size of the queries we are going to make, find the oldest and newest image
-        oldest_image_event_ts = None
-        newest_image_event_ts = 0
-        for image_meta in runtime.image_metas():
+    def find_oldest_newest(self, koji_api):
```
Comment (vfreex) [inline doozer/doozerlib/cli/scan_sources.py:187]: This method does more than its name.

File: doozer/doozerlib/cli/scan_sources.py [hunk 2]
```diff
@@ -3,126 +3,205 @@
...
+                self.oldest_image_event_ts = create_event_ts
+            if create_event_ts > self.newest_image_event_ts:
+                self.newest_image_event_ts = create_event_ts
 
-            if image_meta in changing_image_metas:
+            if image_meta in self.changing_image_metas:
                 continue  # A rebuild is already requested.
 
-            # Request a rebuild if A is a dependent (operator or child image) of B but the latest build of A is older than B.
+            # Request a rebuild if A is a dependent (operator or child image) of B
```
Comment (vfreex) [inline doozer/doozerlib/cli/scan_sources.py:203]: From here, maybe each check can be split into a new method.

**Review by vfreex — COMMENTED** [review]
Generally LGTM. Some nits commented.

File: doozer/doozerlib/cli/scan_sources.py (hunk 2 above)
Comment (locriandev) [reply doozer/doozerlib/cli/scan_sources.py:124]: Done

File: doozer/doozerlib/cli/scan_sources.py (hunk 1 above)
Comment (locriandev) [reply doozer/doozerlib/cli/scan_sources.py]: I kept the same name just for the part that checks build date. Moved the rest to a better place, with more and shorter methods

**Review by vfreex — COMMENTED** [review]
/lgtm

**Pull request merged**